docs/DeveloperGuide.md

```diff
@@ -1,34 +1,330 @@
+   
+  2. Event : Specialized task class used to model events.  
+  
+  3. Deadline : Specialized task class used to model a deadline.   
+   
+  4. TaskType : Enumeration class used to denote the various task types.   
+  
+  5. TaskList : Container class used to store list of tasks and handle related operations. 
+   
+  6. SchedulableTask : Class used to model a task which is scheduled based on user's requirements.  
+  
+  7. TaskComparator : Contains a custom comparator used to compare two schedulable tasks based on their 
+  numberOfDaysLeft attribute.  
+  
+  8.  TaskScheduler : Class used to check for feasibility and schedule a list of tasks based on user's requirements.     
+     
+
+[comment]: # (@@author NizarMohd)
+
+### 2.3. Study Area Component  
+   
+ The Study Area component depends on 3 other components,  
+   
+  1. Command Component - The Command component issues instructions for the Study Area component to execute based on the 
+  user's input. 
+  
+  2. UI Component - The UI component is used to display the results (and related exceptions) of the commands issued to 
+  the Study Area Component.
+  
+  3. ResourceLoader Component - The ResourceLoader component is used to load data of all existing Study Areas in NUS 
+  into text files when the User first run the software. Eventually, data will be referred from the created text file.   
+  
+<div style="text-align:center; font-size: x-small">Object diagram of classes under Study Area Component</div>
+
+ ![Study Area Component](images/StudyAreaObjectDiagram.png)  
+   
+ The Study Area component contains 3 separate classes. They are as follows:  
+  
+   1. Dictionary : Class to map user input to specific terms used in StudyAreaList.
+    
+   2. StudyArea : Class that is used to model Study Areas. 
+   
+   3. StudyAreaList : Class that handles the list of available Study Areas based on User input.  
+        
+## 3. Implementation
+
+### 3.1. Scheduling Tasks 
+
+
+[comment]: # (@@author GanapathySanathBalaji)
+
+#### 3.1.1 Implementation 
+   Inorder to schedule tasks based on the user's requirement a separate SchedulableTask class was created. 
+ The user's requirements (Name, Time to complete it, Deadline) are captured for each of the tasks to be scheduled. 
+ The requirements captured are stored in the SchedulableTask object. 
+ Then, the TaskScheduler object finds the optimum schedule based on the user's requirements using the EDF 
+ (Early Deadline First) algorithm.
+    Since, EDF is an optimum algorithm, if it can't find a valid schedule it means that it's impossible to find a valid schedule based on the user's requirement. If a feasible schedule is found it is 
+ displayed, else a message stating that a schedule based on the user's requirements can't be made is displayed.
+ 
+	
+   The following sequence diagrams explain how tasks are scheduled.
+	
+ ![Overall Sequence Diagram](images/Schedule_Overall.png)
```

Comment:
You might want to explain what is the "ref" frame since it is not included in our syllabus.